### ISSUE TITLE
RVD Error: riparian_values dictionary failing on huc 10190010

### ISSUE THREAD

**Issue by MattReimer** [Riverscapes/riverscapes-tools#148]
This one looks different than #140 

```
2021-01-14 22:09:53:: [debug] [VectorBase] No layer named "ThiessenPolygonsRaw" found
2021-01-14 22:09:53::         PROGRESS: 0 / 48,583  (0%)  Saving /usr/local/data/rvd/10190010/rvd/intermediates/intermediates.gpkg/ThiessenPolygonsRaw     Ellapsed: 0s
2021-01-14 22:09:55:: Finished: 48,583    Saving /usr/local/data/rvd/10190010/rvd/intermediates/intermediates.gpkg/ThiessenPolygonsRaw     
2021-01-14 22:10:00:: [debug] [DatasetRegistry] Dataset closed: /usr/local/data/rvd/10190010/rvd/intermediates/intermediates.gpkg
2021-01-14 22:10:00:: [info] [RVD] Loading Existing and Historic Vegetation Rasters
2021-01-14 22:10:00:: [info] [MakeDir] Folder not found. Creating: /usr/local/data/rvd/10190010/rvd/Intermediates
2021-01-14 22:10:00:: [error] [RVD] int() argument must be a string, a bytes-like object or a number, not 'NoneType'
2021-01-14 22:10:00:: Traceback (most recent call last):
2021-01-14 22:10:00::   File "/usr/local/src/riverscapes-tools/packages/rvd/rvd/rvd.py", line 560, in main
2021-01-14 22:10:00::     rvd(args.huc,
2021-01-14 22:10:00::   File "/usr/local/src/riverscapes-tools/packages/rvd/rvd/rvd.py", line 225, in rvd
2021-01-14 22:10:00::     vegetation["HISTORIC"] = load_vegetation_raster(prj_historic_path, False, output_folder=os.path.join(output_folder, 'Intermediates'))
2021-01-14 22:10:00::   File "/usr/local/src/riverscapes-tools/packages/rvd/rvd/lib/load_vegetation.py", line 126, in load_vegetation_raster
2021-01-14 22:10:00::     riparian_array = np.vectorize(riparian_values.get)(raw_array)
2021-01-14 22:10:00::   File "/usr/local/venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2108, in __call__
2021-01-14 22:10:00::     return self._vectorize_call(func=func, args=vargs)
2021-01-14 22:10:00::   File "/usr/local/venv/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2195, in _vectorize_call
2021-01-14 22:10:00::     res = array(outputs, copy=False, subok=True, dtype=otypes[0])
2021-01-14 22:10:00:: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

The error is on this line:
https://github.com/Riverscapes/riverscapes-tools/blob/master/packages/rvd/rvd/lib/load_vegetation.py#L125

and it deals with `riparian_values` dictionary for the raster `10190010/inputs/historic_veg.tif`

I did see some weird things in the json like both the nodata value and -9999 but I'm not sure if these are intentional or not:

```
{"-9999": 0, "11": 0, "12": 0, "31": .... "32767.0": 0}'
```

Normally we mask when reading rasterio to make sure nodata values don't get processed
```
raw_array = raster.read(1, masked=True)
```

Again, this might be deliberate so I can't say.

### Pro tip:

if you don't feel like running VBET and RSContext you can just steal the ones that cybercastor ran

1. Step 1. Make sure your `.riverscapes` file has an entry in its dictionary called:

``` json
{
    "AnabranchDEV": {
      "url": "https://wXXXXXXXXXXX5.execute-api.us-west-2.amazonaws.com/staging/api",
      "id": "Anabranch",
      "apiUser": "YOUR@EMAIL",
      "apiPass": "XXXXXXXXXXXXXXXXX",
      "dataroot": "SOMEPATH"
    }
}
```

2. Download just what you need to run this command
``` bash
rscli download c:\whatever\rs_context\10190010 --type "RSContext" --meta "huc8=10190010" --tags "JAN13" --file-filter "(hydrology|vegetation)" --verbose --program "AnabranchDEV"

# Go get vbet result for this to work
rscli download c:\whatever\vbet\10190010 --type "VBET" --meta "huc8=10190010" --tags "JAN13" --file-filter "vbet.gpkg" --verbose --program "AnabranchDEV"
```

### Of Note:

* `--file-filter` is a Regex that selects which files to get.
* `--tags` makes sure that you only get the most recent runs
* `--verbose` lots of extra messaging

**Comment by MattReimer** [comment]
## NOTE: 10190003 affected too.

**Comment by KellyMWhitehead** [comment]
Raster value 2718 in Historic Vegetation is not in the xml that ships with the landfire raster (these do not appear in the raster attribute table in esri). 

Looks like there are 2 dozen or so additional classes appended to [this csv on the landfire site](https://www.landfire.gov/CSV/BPS_09172020.csv). These classes are also not present in our [csv](packages/rvd/database/data/VegetationTypes.csv) used to generate our db.

Proposal:
1) Add and update our csv with the new classes, and include the Physiognomy and land use classes. These will now be included in the VegetationTypes table in the output gpkg.
2) Use this csv/database table to populate the reclassification instead of the xml.

Pros: one source of truth, not dependent on xml that esri could change.
Con: we need to update csv this with future landfire versions, but the reality is we would have had to do this anyway.

**Comment by MattReimer** [comment]
My suggestion, regardless of the choice above would be to fail gracefully and spit out a good warning so it doesn't break next time there's a class missing.

We should probably add the classes too though.

**Comment by KellyMWhitehead** [comment]
Fixed in 545b49f1c65ea1d457a9c1ef0aa9f7d7e5c5343a

- RVD now pulls landfire raster values from the VegetationTypes.csv (which is loaded into the gpkg) instead of the xml file that ships with the raster. 
- Updated data dictionary to include missing values
- RVD will now throw an error if it encounters a raster value not in the current data dictionary.
